integration/spark/src/main/scala/org/apache/spark/sql/secondaryindex/optimizer/CarbonSecondaryIndexOptimizer.scala

```diff
@@ -58,6 +58,16 @@ object NodeType extends Enumeration {
  */
 class CarbonSecondaryIndexOptimizer(sparkSession: SparkSession) {
 
+  // to store the sort node per query
+  var sortNodeForPushDown: Sort = _
+
+  // to store the limit literal per query
+  var limitLiteral : Literal = _
+
+  // by default do not push down notNull filter,
+  // but for orderby limit push down, push down notNull filter also. Else we get wrong results.
+  var pushDownNotNullFilter : Boolean = _
```

Comment:
Why not keep these as local variables in transformFilterToJoin and pass to rewritePlanForSecondaryIndex()?

Reply:
because too many functions need to change to pass arguments. 

I used default arguments and changed required places now. so it is local variable now